bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -280,14 +280,13 @@ void ReplicaImp::onMessage<ReplicaAsksToLeaveViewMsg>(ReplicaAsksToLeaveViewMsg
 }
 
 bool ReplicaImp::checkSendPrePrepareMsgPrerequisites() {
-  if (!isCurrentPrimary()) {
-    LOG_WARN(GL, "Calling checkSendPrePrepareMsgPrerequisites in non primary replica!");
-    return false;
-  }
+  ConcordAssert(isCurrentPrimary());
```

Comment:
I cannot agree with leaving this as an assertion. There could be a race between PreProcessor that still thinks it is a primary and ReplicaImp that have changed a primary a moment later. To be on the safe side we need to just ignore the request.

Reply:
but Yulia, in what case would a non-primary send a pre-prepare?

Reply:
I can explain to you offline

Reply:
ok @HristoStaykov , sorry for confusing you... Let's replace the assert with an if statement, as you had done originally. I'll talk to Yulia offline to better understand the reasoning behind this.